pkg/test/framework/components/echo/config.go

```diff
@@ -71,6 +71,9 @@ type Config struct {
 	// Headless (k8s only) indicates that no ClusterIP should be specified.
 	Headless bool
 
+	// StatefulSet indicates that
+	StatefulSet bool
```

Comment:
could merge with headless? Otherwise very confusing

Reply:
I think they may be two distinct cases. You can have headless service that is not statefulset? But maybe we can/should just merge it for simplicity

Reply:
I think they're pretty distinct cases.

Reply:
Yep, something related. Normally statefulset belong to a subset of headless service